### DISCUSSION TITLE
Function Joins

### DISCUSSION THREAD

**Discussion by Walterhart** [supabase/supabase#10571] [Questions]
This about join 2 tables using a function. So I have 2  tables Netflix and Hulu. How can I make a function that  join the 2 tables and display all videos? Note: some data like season has  null values. Also they both have same type of attributes,
![image](https://user-images.githubusercontent.com/70658679/204221798-57e04c57-5cd8-4d29-94f7-d671d1be0827.png)
![image](https://user-images.githubusercontent.com/70658679/204221951-b928d3b2-439c-4c97-ab16-739d2966c839.png)

This the way I am trying currently but I dont know returns type I should use and do not know if I  query the function correctly.
![image](https://user-images.githubusercontent.com/70658679/204224045-b1f4628f-6bd3-4226-8ef3-6f351b3525e0.png)

**Comment by steve-chavez** [comment]
Should be like:

```sql
RETURNS TABLE(id hulu.id%TYPE, name hulu.name%TYPE, other netflix.other%TYPE) -- all the columns you've included in the SELECT
```

Are you using supabase-js? If so, you could join the tables with `select()`, and avoid the function. See https://supabase.com/docs/reference/javascript/select#query-foreign-tables
